apps/recorder/widget.js

```diff
@@ -256,10 +256,34 @@
     options = options||{};
     if (isOn && !settings.recording) {
       var date=(new Date()).toISOString().substr(0,10).replace(/-/g,""), trackNo=10;
-      function getTrackFilename() { return "recorder.log" + date + trackNo.toString(36) + ".csv"; }
-      if (!settings.file || !settings.file.startsWith("recorder.log" + date)) {
-        // if no filename set or date different, set up a new filename
-        settings.file = getTrackFilename();
+      function nextTrackFilename() { return "recorder.log" + date + trackNo.toString(36) + ".csv"; }
+      function mostRecentFile() {
+        const logs = require("Storage").list(/^recorder\.log/);
+
+        if(logs.length > 0){
+          logs.sort();
+          const latest = logs[logs.length-1].replace("\1", "");
+          const f = require("Storage").open(latest, "r");
+          const timeIdx = f.readLine().replace("\n", "").split(",").indexOf("Time");
+          if(timeIdx >= 0){
+            let last, tmp;
+            while(tmp = f.readLine())
```

Comment:
there's a lint warning here - I think you can just use`while((tmp = f.readLine()))` to fix it